### ISSUE TITLE
Add new accuracy metric for machine learning applications

### ISSUE THREAD

**Issue by nicola-bastianello** [team-decent/decent-bench#189]
In machine learning applications we very often care about the accuracy of the trained model (rather than/together with optimality). The idea then is to add a new metric that computes the accuracy of a trained model.

The following should be accomplished:

- [ ] add a `forward` method to data-driven costs, to allow evaluation of the trained model on a set of data
- [ ] customize the way accuracy is displayed in tables (as percentage not scientific notation); new issue will be opened

**Comment by Simpag** [comment]
There is an issue with different use cases. For example if we have a classification task we will have an output vector of probabilities and the forward method should output the argmax of these, but if we have some other task such as density estimation the output of forward should not be the argmax but the output vector of the model. Since both of these tasks share the same cost-function (lets say a neural network one) we can't hardcode the final activation. Should we therefore add another parameter to either the constructor or the forward method which is something like `forward_activation/output_activation`? This would allow us to pass `torch.argmax` for classification tasks and nothing for the density task. We could make some convenient aliases for common usecases such as `classification=argmax`, `raw=None`, `probability=sigmoid` etc.
